lib/mistral_ai_summary_example/summary_settings_page.dart

```diff
@@ -0,0 +1,197 @@
+import 'package:flutter/material.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_summary_example/summary_settings.dart';
+import 'package:mistralai_client_dart/mistralai_client_dart.dart';
+
+class SettingsWidget extends StatefulWidget {
+  const SettingsWidget({
+    required this.initialSettings,
+    required this.mistralAIClient,
+    super.key,
+  });
+
+  final SummarySettings initialSettings;
+  final MistralAIClient mistralAIClient;
+
+  @override
+  State<SettingsWidget> createState() => _SettingsWidgetState();
+}
+
+class _SettingsWidgetState extends State<SettingsWidget> {
+  String model = 'mistral-small';
+  double temperature = 0.7;
+  double topP = 1;
+  int? maxTokens;
+  bool safePrompt = false;
+  int? randomSeed;
+
+  final TextEditingController modelController = TextEditingController();
+  final TextEditingController maxTokensController = TextEditingController();
+  final TextEditingController randomSeedController = TextEditingController();
```

Comment:
Remember to `dispose` of those controllers.

Reply:
Resolved in [f6093b7](https://github.com/nomtek/flutter_ai_examples/pull/3/commits/f6093b70d8fb4ca369c03db88050a992106ed531)